httputil/client.go

```diff
@@ -21,16 +21,130 @@ package httputil
 
 import (
 	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
 	"net/http"
 	"net/url"
+	"path/filepath"
 	"time"
+
+	"github.com/snapcore/snapd/logger"
 )
 
+// CertData contains the raw data of a certificate and the origin of
+// the cert, this is usually a file path on disk and is just used
+// for error reporting.
+type CertData struct {
+	Raw    []byte
+	Origin string
+}
+
+// ExtraSSLCerts is an interface that provides a way to add extra
+// SSL certificates to the httputil.Client
+type ExtraSSLCerts interface {
+	Certs() ([]*CertData, error)
+}
+
+// ExtraSSLCertsFromDir implements ExtraSSLCerts and provides all the
+// pem encoded certs from the given directory.
+type ExtraSSLCertsFromDir struct {
+	Dir string
+}
+
+// Certs returns a slice CertData or an error.
+func (e *ExtraSSLCertsFromDir) Certs() ([]*CertData, error) {
+	extraCertFiles, err := filepath.Glob(filepath.Join(e.Dir, "*.pem"))
```

Comment:
sorry, I missed the commit with the change to e.Dir

Reply:
No worries, thanks for the thoughtful review!